tests/ui/associated-types/ice-issue-137508.rs

```diff
@@ -0,0 +1,10 @@
+trait Tr {
+    type Item;
+}
+
+fn main() {
+    let _: dyn Tr + ?Foo();
```

Comment:
This doesn't have to do with parenthesized syntax. This also ICEs: `+ ?Foo<Assoc = ()>`.

Please edit the test name so it's more clear what it's testing, add the issue number as a comment like:

`// Fix for <https://github.com/issues....>`

Reply:
Sorry, I forgot to mention explicitly:

Change the test to use `+ ?Foo<Assoc = ()>` rather than the parentheses syntax.

Reply:
Ok. Thanks.